### PR TITLE
[AMORO-4229] Refactor synchronizing Hive tables process via ProcessFactory plugin

### LINKED ISSUE #4229: [Subtask]: Refactor synchronizing Hive tables process via ProcessFactory plugin
### Search before asking

- [x] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Parent issue

#4202 

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangwl9**
## Why are the changes needed?

Close #4229.

## Brief change log

-

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-ams/src/main/java/org/apache/amoro/server/process/iceberg/IcebergProcessFactory.java [hunk 1]
```diff
@@ -275,6 +293,14 @@ private Optional<TableProcess> triggerAutoCreateTag(TableRuntime tableRuntime) {
     return Optional.of(new TagsAutoCreatingProcess(tableRuntime, localEngine));
   }
 
+  private Optional<TableProcess> triggerHiveCommitSync(TableRuntime tableRuntime) {
+    if (localEngine == null) {
+      return Optional.empty();
+    }
+
+    return Optional.of(new HiveCommitSyncProcess(tableRuntime, localEngine));
```
Comment (zhoujinsong) [inline amoro-ams/src/main/java/org/apache/amoro/server/process/iceberg/IcebergProcessFactory.java:301]: `triggerHiveCommitSync` is missing a table format check before creating the process.

The `IcebergProcessFactory.formats` list includes `ICEBERG`, `MIXED_ICEBERG`, and `MIXED_HIVE`, so this trigger will be invoked for all three formats. Without an early guard, a `HiveCommitSyncProcess` is created and submitted to the thread pool for every table, only to discover inside `run()` that the table is not a Hive table and return early — wasted scheduling overhead on every cycle.

For comparison, `triggerAutoCreateTag` already does this correctly:
```java
if (localEngine == null
    || tableRuntime.getFormat() != TableFormat.ICEBERG
    || ...) {
  return Optional.empty();
}
```

Suggested fix — filter at the trigger layer:
```java
private Optional<TableProcess> triggerHiveCommitSync(TableRuntime tableRuntime) {
  if (localEngine == null
      || tableRuntime.getFormat() != TableFormat.MIXED_HIVE) {
    return Optional.empty();
  }
  return Optional.of(new HiveCommitSyncProcess(tableRuntime, localEngine));
}
```

The `TableTypeUtil.isHive()` guard inside `HiveCommitSyncProcess.run()` can remain as a defensive check, but the trigger layer should be the first gate.

Comment (zhangwl9) [reply amoro-ams/src/main/java/org/apache/amoro/server/process/iceberg/IcebergProcessFactory.java:301]: fixup

**Review by zhoujinsong — APPROVED** [review]
LGTM.

Thanks for the work!
